### PR TITLE
Products cart functionality

### PR DESCRIPTION
**Pull request by mohitdhatrak**
Deploy preview: https://deploy-preview-7--onsport.netlify.app/

Change log: Added cart related functionalities, handled edge cases. 

Screenshots:

image

image

### CODE REVIEW

File: src/pages/Cart/components/CartProductCard.css [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+.cart-quantity-button {
+    /* flex properties */
+    display: flex;
+    align-items: center;
+}
+
+.cart-quantity-icons {
+    font-size: 1.1rem !important;
+    color: #475569;
```
Comment (akash-1318) [inline src/pages/Cart/components/CartProductCard.css:93]: In case you forgot please use color variables.

**Review by akash-1318 — COMMENTED** [review]
Just added a simple suggestion other then that the code is well structured with proper easy to understand comments and good naming convention , great work .

File: src/pages/Cart/components/CartProductCard.css [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    border-right: 1px solid #808080;
+}
+
+/* .cart-card-button:hover {
+    background-color: var(--cart-card-button-hover-color);
+}
+
+.cart-card-button:focus {
+    outline: var(--cart-card-button-focus-outline-color) solid 2px;
+} */
```
Comment (Gautam2305) [inline src/pages/Cart/components/CartProductCard.css]: remove unnecessary comments from the code

File: src/pages/Cart/components/CartProductCard.css [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    width: 65%;
+    font-size: 0.9rem;
+    font-weight: 500;
+    font-family: "Roboto", sans-serif;
+    cursor: pointer;
+    background-color: transparent;
+    border: none;
+    padding: 0.5em 1em;
+    text-transform: uppercase;
+    color: #808080;
```
Comment (Gautam2305) [inline src/pages/Cart/components/CartProductCard.css:148]: use CSS variables instead of color codes everywhere.

**Review by Gautam2305 — APPROVED** [review]
suggested some changes. Rest all looks good
Things i liked about your code: 
- usage of modal classes
- consistent code throughout
- ES6+ usage in your code with clean JS

**Pull request merged**